transport/src/main/java/io/scalecube/transport/TransportImpl.java

```diff
@@ -195,6 +195,36 @@ public NetworkEmulator networkEmulator() {
                     "Failed to send {} to {}, cause: {}", message, address, ex.toString()));
   }
 
+  public Mono<Message> requestResponse(final Message request, Address address) {
+    return Mono.create(
+        sink -> {
+          Objects.requireNonNull(request, "request must be not null");
+          Objects.requireNonNull(request.correlationId(), "correlationId must be not null");
+
+          listen()
+              .filter(resp -> resp.correlationId() != null)
+              .filter(resp -> resp.correlationId().equals(request.correlationId()))
+              .take(1)
+              .subscribe(
+                  msg -> {
+                    sink.success(msg);
+                  },
+                  error -> {
+                    sink.error(error);
+                  });
```

Comment:
Add another block where you would complete sink  without success or error, just complete without result.